backend/docs/introducing_new_tools.md

```diff
@@ -0,0 +1,129 @@
+
+Now search for `add_label_element` method in the same file. As you can see this method handless adding new label elements by the tool that the label was made with. Please add your tool and the method that will be responsible for creating such label.
+
+If your tool requires some additional validation. Please write the validation method and add it to `validate_label_payload` in `medtagger/api/scans/business.py`.
+
+#### Step 4.3 REST Service
+
+As you can see in `medtagger/api/scans/service_rest.py` we do already have endpoint that adds new label for given scan. What is now important is to define a schema that will ensure that whatever JSON representation of a label made with your tool is correct.
+
+To define a schema proceed to `medtagger/api/scans/serializers.py`. As you can see we have defined a schema for every `LabelElement` and we are actually telling that our new `LabelElement` can be one of the following types:
+
+```python
+elements_schema = {
+    'type': 'array',
+    "items": {
+        "type": "object",
+        'oneOf': [
+            {'$ref': '#/definitions/rectangular_label_element_schema'},
+            {'$ref': '#/definitions/brush_label_element_schema'},
+            {'$ref': '#/definitions/point_label_element_schema'},
+            {'$ref': '#/definitions/chain_label_element_schema'},
+        ],
+    },
+```
+Below is and example of one of the schemas for label made with rectangle tool:
+
+```python
+'rectangular_label_element_schema': {
+    'properties': {
+        'x': {'type': 'number', 'minimum': 0.0, 'maximum': 1.0},
+        'y': {'type': 'number', 'minimum': 0.0, 'maximum': 1.0},
+        'width': {'type': 'number', 'minimum': 0.0, 'maximum': 1.0},
+        'height': {'type': 'number', 'minimum': 0.0, 'maximum': 1.0},
+        'slice_index': {'type': 'integer'},
+        'tag': {'type': 'string'},
+        'tool': {'type': 'string', 'pattern': 'RECTANGLE'},
+    },
+    'required': ['x', 'y', 'width', 'height', 'slice_index', 'tag', 'tool'],
+    'additionalProperties': False,
+},
+```
+Your job would be now to:
+
+1. Create your own definition of the schema,
+2. Write down every property that your `new_tool_label_element` should have,
+3. Provide additional information about properties (type, range etc.),
+4. Make sure to define which of the properties are required,
+5. Add your new definition to the `oneOf` section of the `elements_schema`.
+
+If you want more info about how to create such schemas, visit [this](https://json-schema.org/understanding-json-schema/index.html) page
+
+### Step 5. Tests
+
+To make sure that all of this actually works, we need to write some tests! Proceed to `tests/functional_tests/test_adding_new_label.py`. This file holds tests for creating label for every tool that is supported by MedTagger. As you can see all of the tests have following steps:
```

Comment:
"Proceed to `tests/functional_tests/test_adding_new_label.py`. This file holds tests..." -> "Go to the `tests/functional_tests/test_adding_new_label.py` which contains all tests..."